includes/gateway/class-omise-payment-promptpay.php

```diff
@@ -0,0 +1,207 @@
+							}, 1000);
+					}
+
+					let elementRefreshButton         = $( '#omise-offline-payment-refresh-status' ),
+						elementOmiseOfflineDetails   = $( '#omise-offline-additional-details' ),
+						elementTimeoutMessage        = $( '#omise-offline-payment-timeout' ),
+						elementPaymentResult         = $( '#omise-offline-payment-result' ),
+						validatePaymentResultCount   = 0;
+						validatePaymentResultTimerId = null,
+						intervalTime                 = 10000; // 10 seconds
+						maxIntervalTime              = 600000; // 10 minutes
+						validatePaymentResult        = function() {
+							validatePaymentResultCount++;
+
+							$.ajax({
+								type: 'POST',
+								url: '<?php echo admin_url( 'admin-ajax.php' ); ?>',
+								dataType: 'json',
+								data: {
+									'action': 'fetch_order_status',
+									'order_id': '<?php echo $order; ?>'
+								},
+								success: function( response ) {
+									if ( 'processing' === response.data.order_status ) {
+										elementOmiseOfflineDetails.fadeOut();
+										elementPaymentResult.html('<h4><?php echo __( "Payment successful", "omise" ); ?></h4><p><?php echo __( "Thank you. Your order has been paid.", "omise" ); ?></p>').fadeIn();
+										clearInterval( validatePaymentResultTimerId );
+										return;
+									} else if ( 'failed' === response.data.order_status ) {
+										elementOmiseOfflineDetails.fadeOut();
+										elementPaymentResult.html('<h4><?php echo __( "Payment failed", "omise" ); ?></h4><p><?php echo __( "Please try placing an order with a different payment method.", "omise" ); ?></p>').fadeIn();
+										clearInterval( validatePaymentResultTimerId );
+										return;
+									}
+								},
+								error: function( response ) { console.log(response); },
+								complete: function (response ) {
+									if ( validatePaymentResultCount * intervalTime >= maxIntervalTime ) {
+										elementTimeoutMessage.fadeIn();
+										console.log('stop');
+										return;
+									}
+								}
+							});
+						};
+
+					elementRefreshButton.click( function() {
+						validatePaymentResultCount   = 0;
+						validatePaymentResultTimerId = initializeTimer();
+
+						elementTimeoutMessage.fadeOut();
+					} );
+
+					$( document ).ready( function() {
+						validatePaymentResultCount   = 0;
+						validatePaymentResultTimerId = initializeTimer();
+					});
+				})(jQuery);
+			</script>
+		<?php elseif ( 'email' === $context ) : ?>
+			<div>
+				<?php
+				echo sprintf(
+					wp_kses(
+						__( 'Please scan the QR code and pay before: <strong>%1$s %2$s</strong>', 'omise' ),
+						array( 'strong' => array() )
+					),
+					wc_format_datetime( $expires_datetime, wc_date_format() ),
+					wc_format_datetime( $expires_datetime, wc_time_format() )
+				);
+				?>
+			</div>
+			<p><a href="<?php echo $qrcode; ?>"><?php echo __( 'Click this link to display the QR code', 'omise' ); ?></a></p>
+		<?php endif;
+	}
+}
```

Comment:
I'm not strong at Javascript. If there is any suggestion to improve or to make the code more effective, please feel free to let me know 🙇